reinforcement_learning/common/sagemaker_rl/ray_launcher.py

```diff
@@ -231,6 +240,47 @@ def save_checkpoint_and_serving_model(self, algorithm=None, env_string=None, con
         # To ensure SageMaker local mode works fine
         change_permissions_recursive(INTERMEDIATE_DIR, 0o777)
         change_permissions_recursive(MODEL_OUTPUT_DIR, 0o777)
+        
+    def set_up_checkpoint(self, config=None):
+        try:
+            checkpoint_dir = config['training']['restore']
+            print("Found checkpoint dir in user config.")
+        except KeyError:
+            pass
+
+        try:
+            # validate files in checkpoint channel and set restore dir in ray config
+            validation = 0
+            if len(os.listdir(CHECKPOINT_DIR)) is not (2 or 3):
+                raise RuntimeError("Unexpected files in checkpoint dir.",
+                                   "Please check ray documents for the correct checkpoint format.")
+            checkpoint_dir = ""
```

Comment:
A little bit confusing, is `CHECKPOINT_DIR ` the directory for checkpoint or `checkpoint_dir`? Seems `checkpoint_dir` is the file path of checkpoint?

Reply:
Yeah, the transition from downloading the checkpoint files to how Ray can actually use them is a bit tricky. When customer provides a checkpoint s3 location, say, `s3://path/to/ray/results/checkpoint_100`, the files under `checkpoint_100` get downloaded to `CHECKPOINT_DIR`.  There will only be 2 files (`checkpoint-100` and `checkpoint-100.tune_metadata`) for ray version >= 0.6.5, and 3 files (another `checkpoint-100.extra_data`) for the older version. 

Then for ray to recognize them, `restore` should be set to `CHECKPOINT_DIR/checkpoint-100`, instead of `CHECKPOINT_DIR`. The logic here is to extrat `checkpoint-100` and attach it to `CHECKPOINT_DIR`